src/pybind/mgr/ssh/module.py

```diff
@@ -1094,3 +1094,53 @@ def remove_rgw(self, name):
 
     def update_rgw(self, spec):
         return self._update_service('rgw', self.add_rgw, spec)
+
+    def add_rbd_mirror(self, spec):
+        if not spec.placement.nodes or len(spec.placement.nodes) < spec.count:
+            raise RuntimeError("must specify at least %d hosts" % spec.count)
+        daemons = self._get_services('rbd-mirror')
+        results = []
+        num_added = 0
+        for host in spec.placement.nodes:
+            if num_added >= spec.count:
+                break
+            daemon_id = self.get_unique_name(daemons)
+            self.log.debug('placing rbd-mirror.%s on host %s' % (daemon_id,
+                                                                 host))
+            results.append(
+                self._worker_pool.apply_async(self._create_rbd_mirror,
+                                              (daemon_id, host))
+            )
+            # add to daemon list so next name(s) will also be unique
+            sd = orchestrator.ServiceDescription()
+            sd.service_instance = daemon_id
+            sd.service_type = 'rbd-mirror'
+            sd.nodename = host
+            daemons.append(sd)
+            num_added += 1
+        return SSHWriteCompletion(results)
+
+    def _create_rbd_mirror(self, daemon_id, host):
+        ret, keyring, err = self.mon_command({
+            'prefix': 'auth get-or-create',
+            'entity': 'client.rbd-mirror.' + daemon_id,
+            'caps': ['mon', 'allow profile rbd-mirror',
+                     'osd', 'profile rbd'],
```

Comment:
@dillaman One thing I noticed while doing this: the mon cap is  'allow profile foo' while the osd cap is 'profile foo'. Actually, it looks like the 'allow' is optional for the mon.. works both ways.  But I wonder if we should consistify this?

Reply:
Yeah -- "allow profile" never made much sense to me (i.e. what would "deny profile" imply?).  In the RBD docs we always just use 'profile XYZ' throughout.

Reply:
Yeah... in reality *everything* in an ``allow ...`` since it's strictly additive. Oh well

Reply:
In theory one day a "deny XYZ" cap could be added -- but just not a "deny profile XYZ".

Reply:
good point!